hack/test-port-forwarding.pl

```diff
@@ -20,8 +20,24 @@
 use Socket qw(inet_ntoa);
 use Sys::Hostname qw(hostname);
 
+# check for nc
+sub check_netcat_installed {
+    my $nc_path = `which nc 2>/dev/null`;
+    chomp $nc_path;
+    unless ($nc_path) {
+        die "Error: 'nc' (netcat) is not installed on the host system.\n" .
+            "Please install netcat to run this test script:\n" .
+            "  - On macOS: brew install netcat\n" .
+            "  - On Ubuntu/Debian: sudo apt-get install netcat\n" .
+            "  - On RHEL/CentOS: sudo yum install nmap-ncat\n";
+    }
+}
+
 my $instance = shift;
 
+# Check for netcat before proceeding
+check_netcat_installed() unless -f $instance;
```

Comment:
Why not move the check after the `if (-f $instance)` block below, so you don't need to repeat the check?

And given that the whole file is just one long script, I think the code should just be inlined and not put into a function.

Reply:
Done 😄